src/daemon/control.rs

```diff
@@ -850,6 +854,194 @@ pub fn watchtowers_status(revaultd: &RevaultD) -> Vec<ServerStatus> {
     watchtowers
 }
 
+/// get_history retrieves a limited list of events which occured between two given dates.
+pub fn get_history<T: BitcoindThread>(
+    revaultd: &RevaultD,
+    bitcoind_conn: &T,
+    start: u64,
+    end: u64,
+    limit: u64,
+    kind: Vec<HistoryEventKind>,
+) -> Result<Vec<HistoryEvent>, RpcControlError> {
+    let db_path = revaultd.db_file();
+    // All vaults which have one transaction (either the funding, the canceling, the unvaulting, the spending)
+    // inside the date range are retrieved.
+    // This list might include vaults that were consumed again outside the range.
+    let vaults = db_vaults_with_txids_between_date(&db_path, start, end, limit)?;
+
+    let mut spends: HashMap<Txid, Vec<&DbVault>> = HashMap::new();
+    let mut events: Vec<HistoryEvent> = Vec::new();
+
+    for vault in &vaults {
+        if kind.contains(&HistoryEventKind::Deposit)
+            && vault.status != VaultStatus::Unconfirmed
```

Comment:
Because the transaction must be confirmed, the vault even as a change deposit should be funded +1

Reply:
Hmmm. Don't you think it could/should be checked for in the SQL request then?

Also `s/calcul/compute/` :fr: :p

Reply:
Not really, because the vault is needed for the computation of the fees of the cancel transaction even if it is unconfirmed
```rust
            let change_amount = vaults_by_outpoint_txid
                .get(&txid)
                // if the change output did not create a vault because of the dust limit
                // the change_amount is equal to 0.
                .map_or(0, |vault| vault.amount.as_sat());
```
I could use the derivation_index of the vault canceled to derive the cancel tx and find the change output but then I cannot know if the daemon considered the change output as a vault or a dust output